src/changelog-updater.ts

```diff
@@ -156,34 +156,42 @@ async function parseChangelogForEntry(
   for await (const line of fileStream) {
     contents.push(line)
 
-    // If we have found the line to update, the last line to add the entry after, or have found
-    // a duplicate line, just push the line
-    if (foundLastEntry || foundDuplicateEntry || foundEntryToUpdate) {
+    if (
+      foundLastEntry ||
+      foundDuplicateEntry ||
+      foundEntryToUpdate ||
+      EMPTY_LINE_REGEX.test(line)
+    ) {
+      lineNumber++
       continue
     }
 
-    // Only check the line if we haven't found the entry before
-    if (versionFound && line.startsWith(entryLine)) {
-      foundDuplicateEntry = true
-    } else if (versionFound && line.startsWith(entryLineStart)) {
-      foundEntryToUpdate = true
-      changelogLineNumber = lineNumber
-    }
-
-    if (versionFound && DEPENDENCY_SECTION_REGEX.test(line)) {
-      dependencySectionFound = true
-      changelogLineNumber = lineNumber
-    }
-
-    if (!versionFound && versionRegex.test(line)) {
+    if (versionFound) {
```

Comment:
I'll need to think about some of this. A few of these checks seem redundant to me - we really should only need to know "did we find the version, but not find a `### Dependencies` section and then append to the end of the version, which I think can be done somewhat cleaner.

I'll do some thinking here and see if I can find a simpler way to implement this check with the tests (thank you!) that you've added

Reply:
Ok - I think I understand this now.

My only concern is if changelogs use `-` to denote other additions in other sections, which I guess we should add a test for. If I understand it correctly, that could allow us to put an entry in the wrong section if that section comes after the `### Dependencies` section. 

Am I wrong there?

Reply:
@dangoslen Can you give me an example of the kind of layout you mean and I can certainly add a test case for it?

Reply:
I think a layout like this

```md
## [version]

### Dependencies
- one
- two

### Added
- one
- two
```

I _think_ that layout would mean appending the entry under `### Added`

Reply:
I've added another test case to be certain, but that case is handled here: https://github.com/dangoslen/dependabot-changelog-helper/pull/123/files#diff-ae6bdbbd1f9a828cdbdd3157ce42cf92bbd7379e2ad42928fdbbba1acd1c2d60R171-R173 

If we're "in" the dependencies section, and another header is encountered then it sets `foundLastEntry` so that it stops any further processing.

Reply:
Ah. I'm tracking now. 

Happy to merge! Just add an entry to either `Added` or `Fixed` (up to you) with a description of this change under the `Unreleased` version, and I'll get a `v2.2.0` cut